test/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter_test.cc

```diff
@@ -1608,6 +1639,46 @@ TEST_F(GrpcJsonTranscoderFilterConvertGrpcStatusTest, SkipTranscodingStatusIfBod
   EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_.decodeTrailers(request_trailers));
 }
 
+class GrpcJsonTranscoderFilterConvertGrpcStatusByExtensionTest
+    : public GrpcJsonTranscoderFilterGrpcStatusTest {
+public:
+  GrpcJsonTranscoderFilterConvertGrpcStatusByExtensionTest()
+      : GrpcJsonTranscoderFilterGrpcStatusTest(makeProtoConfig()) {}
+
+private:
+  const envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder
+  makeProtoConfig() {
+    auto proto_config = bookstoreProtoConfig();
+    proto_config.set_convert_grpc_status(true);
+    proto_config.mutable_grpc_status_json_marshaller()->set_name(
+        "envoy.proto_json_marshaller.transcoder_test_proto_json_marshaller");
+    return proto_config;
+  }
+};
+
+TEST_F(GrpcJsonTranscoderFilterConvertGrpcStatusByExtensionTest,
+       TranscodingBinaryHeaderInTrailerOnlyResponse) {
```

Comment:
can we add a integration test for this?

Reply:
You mean grpc_json_transcoder_integration test. Seems unnecessary to me. The json_transcoder_filter_test should already be enough.